### ISSUE TITLE
App asks for too many permissions on install

### ISSUE THREAD

**Issue by Microflow** [Microflow/fforward-android#27]
Are these all 100% necessary for a Reddit reader app?

   - Privacy: Modify or delete SD card contents
   - Read the contents of your SD card
   - Device access: Full network access, network connections (two things)
   - Startup: Run at startup

How do I remove some of these permissions and still keep a working reddit reader app?

Asked here:

https://www.reddit.com/r/androiddev/comments/8tpk9w/weekly_questions_thread_june_25_2018/e1alaid/

**Comment by Microflow** [comment]
This may help:

https://stackoverflow.com/questions/35701792/android-app-not-asking-for-permissions-when-installing

**Comment by Microflow** [comment]
Good primer on requesting permissions:

https://developer.android.com/training/permissions/requesting

**Comment by Microflow** [comment]
https://www.androidauthority.com/android-sdk-tutorial-beginners-634376/

**Comment by Microflow** [comment]
https://stackoverflow.com/questions/24510219/what-is-the-difference-between-min-sdk-version-target-sdk-version-vs-compile-sd

**Comment by Microflow** [comment]
## Compile, Min and Target SDK
 
*0.8% of 1.4 billion devices is a lot*

It is very important to know the difference between all of them. We have to make sure that existing apps built against prior SDKs should not break when the user updates to a new version.

## compileSdkVersion:
It says the Gradle about version of the Android SDK to compile our project. We can use of the new APIs added.
Note: Changing compileSdkVersion will not change runtime behaviour.

It is always recommended that compile with the latest SDK. You will get all the benefits of new compilation checks.

## minSdkVersion:
Basically, it tell which devices an app can be installed on. Actually it is the lower bound for the app. Make sure that our minSdkVersion must be atleast high as our dependencies minSdkVersion Note: 0.8% of 1.4 billion devices is a lot.

## targetSdkVersion:
It is the most important of the three. We cannot apply behaviour changes of or app until we update the target SdkVersion. This allow us to use new APIs. By doing some changes and modifications we can update our app with the latest sdk. We should test properly before updating the targetSdkVersion.

**Comment by Microflow** [comment]
# Graph to Target Versions

https://developer.android.com/about/dashboards/

Eric says: For Android version below 5.0, there still about 14%.

5.0 is fine
and it's easier
If you target 4.X, you may have many compatibility issues

**Comment by hungyanbin** [comment]
BTW, 
That means , you could set minSdkVersion to 21, and targetSdkVersion to the latest version if possible

**Comment by hungyanbin** [comment]
And for permissions:
1. SD card permission: This project can upload or save image. In doing so, you must need READ_EXTERNAL_STORAGE for upload and WRITE_EXTERNAL_STORAGE for save image
2. network access : This is necessary for almost every app
3. Startup: After tracing code, there's Alarm needs to be initialize after boot. 
[Alarms](https://github.com/Microflow/fforward-android/blob/2210513e4ae5ff062892f88811c843ce714c5fa1/src/main/java/org/quantumbadger/redreader/common/Alarms.java), [BootReceiver](https://github.com/Microflow/fforward-android/blob/2210513e4ae5ff062892f88811c843ce714c5fa1/src/main/java/org/quantumbadger/redreader/receivers/BootReceiver.java). But this project do not use any alarm currently, maybe you can remove it If you don't need Alarm.

**Comment by Microflow** [comment]
1. I think we can disable upload or save image for now. The user doesn't
need full Reddit capability. This is mainly a passive reading app.

2. Network access I will keep.

3. Don't need the alarm, so startup/boot can be disabled.

*

I don't want to annoy or worry the user with too many permissions, so I'll
keep it to the bare minimum.
